### PR TITLE
feat: what is a top layer article draft

### LINKED ISSUE #3365: [CSS][DevTools] What is a "Top layer"? [8/9/2022]
* Proposed title: What is a "Top layer"?
* Proposed by: jheyy
* Related areas: CSS,Chrome DevTools
* Content type: Explanation (https://documentation.divio.com/explanation/)
* Deadline: 8/9/2022
* Deadline explanation: We'd like to get this out so it can be referenced from PopUp and SelectMenu articles for DCC
* [Draft](https://docs.google.com/document/d/1SOBCUGo5bbhgLtuPB4j_icE4Nty_nhrRaYC1vDLUwhI)
* Content summary: Introducing the developer community to the concept of the top layer which is used by components and APIs such as popup and dialog.

### PR DESCRIPTION
**Pull request by jh3y**
Fixes #3365 

Changes proposed in this pull request:

- Brief explainer article on the browser's top layer

### CODE REVIEW

File: site/en/blog/what-is-the-top-layer/index.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+---
+layout: "layouts/blog-post.njk"
+title: "What's a \"Top layer\"?"
+description: "With new native components and APIs coming to the platform, what is this \"Top layer\" that they enable us to use?"
```
Comment (rachelandrew) [inline site/en/blog/what-is-the-top-layer/index.md:4]: Please don't use native as per style guide. I'd suggest rewording as, "This post explains the "Top Layer" used in some new components and APIs."

File: site/en/blog/what-is-the-top-layer/index.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+authors:
+  - jheyy
+hero: "image/Dyx9FwYgMyNqy1kMGx8Orz6q0qC3/kogMUEiTkiI5gKqaxrsJ.jpg"
+alt: "Image of layered mountains"
```
Comment (rachelandrew) [inline site/en/blog/what-is-the-top-layer/index.md:8]:
```suggestion
alt: "Layered mountains."
```

File: site/en/blog/what-is-the-top-layer/index.md [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+tags:
+  - css
+  - html
+date: 2022-08-14
```
Comment (rachelandrew) [inline site/en/blog/what-is-the-top-layer/index.md:12]:
```suggestion
date: 2022-08-15
```

File: site/en/blog/what-is-the-top-layer/index.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+---
+
+To cut a long story short, the top layer sits above its related `document` in the browser viewport. And each document has one associated top layer. Thus, elements promoted to the top layer needn't worry about `z-index` or DOM hierarchy. They also get a neat `::backdrop` pseudo-element to play with. The [Fullscreen API](https://fullscreen.spec.whatwg.org/#new-stacking-layer) spec goes further in-depth on this. And [that API](https://developer.mozilla.org/docs/Web/API/Element/requestFullScreen) was a great example of the top layer in use before `dialog` [support came](https://caniuse.com/dialog) along.
```
Comment (rachelandrew) [inline site/en/blog/what-is-the-top-layer/index.md:15]:
```suggestion
The top layer sits above its related `document` in the browser viewport, and each document has one associated top layer. This means that elements promoted to the top layer needn't worry about `z-index` or DOM hierarchy. They also get a neat `::backdrop` pseudo-element to play with. The [Fullscreen API](https://fullscreen.spec.whatwg.org/#new-stacking-layer) spec goes into more details as [Fullscreen](https://developer.mozilla.org/docs/Web/API/Element/requestFullScreen) was a great example of the top layer in use before [`dialog` support](https://caniuse.com/dialog) came along.
```

File: site/en/blog/what-is-the-top-layer/index.md [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+{% Codepen {
+    user: 'web-dot-dev',
+    id: 'xxWYaor',
+    height: 450,
+    tab: 'result'
+  }
+%}
+
+
+But, with new native components and APIs like `dialog` and `pop-up`. Developers won't need to resort to these workarounds. We can promote content to the top layer. UI frameworks allow us to co-locate promoted elements with their component counterparts. But, then often get separated in the DOM when it comes to rendering.
```
Comment (rachelandrew) [inline site/en/blog/what-is-the-top-layer/index.md:38]:
```suggestion
With new components and APIs like `dialog` and `pop-up`, you won't need to resort to these workarounds. You can promote content to the top layer. 

UI frameworks allow us to co-locate promoted elements with their component counterparts. But, they often get separated in the DOM when it comes to rendering.
```

File: site/en/blog/what-is-the-top-layer/index.md [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+
+By using the top layer, promoted elements are where we put them in our source code (for example, a `dialog`). It doesn't matter how many layers down in the DOM the element is. It will get promoted to the top layer and we can inspect it where we expect it to be, co-located with our component HTML. This makes it easier to inspect both the trigger element and the promoted element in the DOM at the same time. Particularly useful if your trigger element is making attribute updates, etc.
```
Comment (rachelandrew) [inline site/en/blog/what-is-the-top-layer/index.md:40]:
```suggestion
By using the top layer, promoted elements are where you put them in your source code (for example, a `dialog`). It doesn't matter how many layers down in the DOM the element is. It will get promoted to the top layer and you can inspect it where you expect it to be, co-located with your component HTML. This makes it easier to inspect both the trigger element and the promoted element in the DOM at the same time. Particularly useful if your trigger element is making attribute updates, for example.
```

**Review by rachelandrew — CHANGES_REQUESTED** [review]
See suggestions inline.

File: site/en/blog/what-is-the-top-layer/index.md (hunk 1 above)
Comment (jh3y) [reply site/en/blog/what-is-the-top-layer/index.md:4]: Yeah, I missed this in my edit, should say "built-in".

File: site/en/blog/what-is-the-top-layer/index.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+title: "What's a \"Top layer\"?"
+description: "With new built-in components and APIs coming to the platform, what is this \"Top layer\" that they enable us to use?"
+authors:
+  - jheyy
+hero: "image/Dyx9FwYgMyNqy1kMGx8Orz6q0qC3/kogMUEiTkiI5gKqaxrsJ.jpg"
+alt: "Layered mountains."
+tags:
+  - css
+  - html
+date: 2022-08-15
```
Comment (rachelandrew) [inline site/en/blog/what-is-the-top-layer/index.md:12]:
```suggestion
date: 2022-08-17
```

**Review by rachelandrew — APPROVED** [review]

File: site/en/blog/what-is-the-top-layer/index.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+title: "What's a \"Top layer\"?"
+description: "With new built-in components and APIs coming to the platform, what is this \"Top layer\" that they enable us to use?"
+authors:
+  - jheyy
+hero: "image/Dyx9FwYgMyNqy1kMGx8Orz6q0qC3/kogMUEiTkiI5gKqaxrsJ.jpg"
+alt: "Layered mountains."
+tags:
+  - css
+  - html
+date: 2022-08-17
```
Comment (rachelandrew) [inline site/en/blog/what-is-the-top-layer/index.md:12]:
```suggestion
date: 2022-08-18
```

**Comment by rachelandrew** [comment]
@jh3y you are good to merge this when you are ready. I've updated the date to today, but you can change that if you like.

File: site/en/blog/what-is-the-top-layer/index.md [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+title: "What's a \"Top layer\"?"
+description: "With new built-in components and APIs coming to the platform, what is this \"Top layer\" that they enable us to use?"
+authors:
+  - jheyy
+hero: "image/Dyx9FwYgMyNqy1kMGx8Orz6q0qC3/kogMUEiTkiI5gKqaxrsJ.jpg"
+alt: "Layered mountains."
+tags:
+  - css
+  - html
+date: 2022-08-18
```
Comment (rachelandrew) [inline site/en/blog/what-is-the-top-layer/index.md:12]:
```suggestion
date: 2022-08-22
```

File: site/en/blog/what-is-the-top-layer/index.md [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+
+{% Codepen {
+    user: 'web-dot-dev',
+    id: 'MWVxmKK',
+    height: 450,
+    tab: 'result'
+  }
+%}
+
+In this demo, you can open a SweetAlert popup and also open a top layer `Dialog`. Open the `Dialog`, and then try opening the SweetAlert popup. You'll see that it appears underneath our top layer element. And the root of our SweetAlert popup is using a `z-index` of 10000 with `position: fixed`.
```
Comment (rachelandrew) [inline site/en/blog/what-is-the-top-layer/index.md:62]:
```suggestion
In this demo, you can open a SweetAlert popup and also open a top layer `<dialog>`. Open the `<dialog>`, and then try opening the SweetAlert popup. You'll see that it appears underneath our top layer element. And the root of our SweetAlert popup is using a `z-index` of 10000 with `position: fixed`.
```

File: site/en/blog/what-is-the-top-layer/index.md [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+
+```css
+.swal-overlay {
+  z-index: 10000;
+  position: fixed;
+}
+```
+
+You don't need to apply any styles to the `Dialog` to make it appear above all other content.
```
Comment (rachelandrew) [inline site/en/blog/what-is-the-top-layer/index.md:71]:
```suggestion
You don't need to apply any styles to the `<dialog>` to make it appear above all other content.
```

File: site/en/blog/what-is-the-top-layer/index.md [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+{% Codepen {
+    user: 'web-dot-dev',
+    id: 'xxWYaor',
+    height: 450,
+    tab: 'result'
+  }
+%}
+
+
+With new built-in components and APIs like `dialog` and `pop-up`, you won't need to resort to these workarounds. You can promote content to the top layer.
```
Comment (rachelandrew) [inline site/en/blog/what-is-the-top-layer/index.md:38]:
```suggestion
With new built-in components and APIs like `<dialog>` and [`<pop-up>`](https://open-ui.org/components/popup.research.explainer), you won't need to resort to these workarounds. You can promote content to the top layer. 
```

File: site/en/blog/what-is-the-top-layer/index.md [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+    tab: 'result'
+  }
+%}
+
+
+With new built-in components and APIs like `dialog` and `pop-up`, you won't need to resort to these workarounds. You can promote content to the top layer. 
+
+UI frameworks allow us to co-locate promoted elements with their component counterparts. But, they often get separated in the DOM when it comes to rendering.
+
+By using the top layer, promoted elements are where you put them in your source code (for example, a `dialog`). It doesn't matter how many layers down in the DOM the element is. It will get promoted to the top layer and you can inspect it where you expect it to be, co-located with your component HTML. This makes it easier to inspect both the trigger element and the promoted element in the DOM at the same time. Particularly useful if your trigger element is making attribute updates, for example. This also has an added benefit for accessibility now that the elements are co-located.
```
Comment (rachelandrew) [inline site/en/blog/what-is-the-top-layer/index.md:42]:
```suggestion
By using the top layer, promoted elements are where you put them in your source code (for example, a `<dialog>`). It doesn't matter how many layers down in the DOM the element is. It will get promoted to the top layer and you can inspect it where you expect it to be, co-located with your component HTML. This makes it easier to inspect both the trigger element and the promoted element in the DOM at the same time. Particularly useful if your trigger element is making attribute updates, for example. This also has an added benefit for accessibility now that the elements are co-located.
```

**Review by rachelandrew — APPROVED** [review]
